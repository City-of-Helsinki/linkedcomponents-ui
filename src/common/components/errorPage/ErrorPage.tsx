import { ClassNames } from '@emotion/react';
import { ButtonVariant, IconArrowLeft } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router';

import { ROUTES } from '../../../constants';
import MainContent from '../../../domain/app/layout/mainContent/MainContent';
import useAuth from '../../../domain/auth/hooks/useAuth';
import useLocale from '../../../hooks/useLocale';
import Button from '../button/Button';
import ErrorTemplate from '../errorTemplate/ErrorTemplate';
import FeedbackButton from '../feedbackButton/FeedbackButton';
import styles from './errorPage.module.scss';

interface Props {
  homePageButtonText?: string | null;
  signInPath: string;
  text: string;
}

const ErrorPage: React.FC<Props> = ({
  homePageButtonText,
  signInPath,
  text,
}) => {
  const locale = useLocale();
  const navigate = useNavigate();
  const { login } = useAuth();
  const { t } = useTranslation();

  const goToHome = () => {
    navigate(`/${locale}${ROUTES.HOME}`);
  };

  const handleSignIn = () => {
    login(signInPath);
  };

  return (
    <ClassNames>
      {({ css }) => (
        <MainContent>
          <ErrorTemplate
            buttons={
              <div className={styles.buttons}>
                <Button
                  fullWidth={true}
                  iconStart={<IconArrowLeft aria-hidden />}
                  onClick={goToHome}
                  type="button"
                  variant={ButtonVariant.Secondary}
                >
                  {homePageButtonText || t('common.goToHome')}
                </Button>
                <Button
                  fullWidth={true}
                  onClick={handleSignIn}
                  type="button"
                  variant={ButtonVariant.Primary}
                >
                  {t('common.signIn')}
                </Button>
              </div>
            }
            text={text}
          >
            <FeedbackButton />
          </ErrorTemplate>
        </MainContent>
      )}
    </ClassNames>
  );
};

export default ErrorPage;
