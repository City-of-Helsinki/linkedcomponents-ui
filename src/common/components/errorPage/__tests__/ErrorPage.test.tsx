import React from 'react';

import {
  configure,
  render,
  screen,
  userEvent,
} from '../../../../utils/testUtils';
import ErrorPage from '../ErrorPage';

configure({ defaultHidden: true });

const renderComponent = () =>
  render(<ErrorPage signInPath="" text={'Lorem ipsum'} />, {
    routes: ['/fi/events'],
  });

test('should render component', () => {
  const { container } = renderComponent();

  expect(container).toMatchSnapshot();
});

test('should route to contact page when clicking feedback link', async () => {
  const user = userEvent.setup();
  const { history } = renderComponent();

  await user.click(screen.getByRole('button', { name: 'Anna palautetta' }));

  expect(history.location.pathname).toBe('/fi/help/support/contact');
});
