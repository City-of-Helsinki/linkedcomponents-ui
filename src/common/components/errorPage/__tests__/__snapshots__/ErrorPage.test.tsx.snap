// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div>
  <main
    class="mainContent"
    id="maincontent"
  >
    <div
      class="container css-0"
    >
      <div
        class="contentWrapper errorPage"
      >
        <div
          class="content"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik icon"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 12v2h-2v-2h2zm0-10v8h-2V6h2z"
                fill="currentColor"
              />
            </g>
          </svg>
          <p>
            Lorem ipsum
          </p>
          <div
            class="buttonsWrapper"
          >
            <div
              class="buttons"
            >
              <button
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS Button-module_fullWidth__31eVK button_hds-button--fullwidth__1_hG9 button css-0"
                type="button"
              >
                <div
                  aria-hidden="true"
                  class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
                >
                  <svg
                    aria-hidden="true"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <rect
                        height="24"
                        width="24"
                      />
                      <polygon
                        fill="currentColor"
                        points="10 5.5 11.5 7 7.5 11 20 11 20 13 7.5 13 11.5 17 10 18.5 3.5 12"
                      />
                    </g>
                  </svg>
                </div>
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  Etusivulle
                </span>
              </button>
              <button
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO Button-module_fullWidth__31eVK button_hds-button--fullwidth__1_hG9 button css-0"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  Kirjaudu sisään
                </span>
              </button>
            </div>
          </div>
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_supplementary__3YKiS button_hds-button--supplementary__GcHcV feedbackButton button css-0"
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h24v24H0z"
                  />
                  <path
                    d="M19 4a4 4 0 013.995 3.8L23 8v8a4 4 0 01-3.8 3.995L19 20h-1c-.988 0-1.846.338-2.617 1.039l-.176.168-.707.707-.707-.707c-.757-.757-1.6-1.149-2.567-1.201L11 20H5a4 4 0 01-3.995-3.8L1 16V8a4 4 0 013.8-3.995L5 4h14zm0 2H5a2 2 0 00-1.995 1.85L3 8v8a2 2 0 001.85 1.995L5 18h6a5.68 5.68 0 013.5 1.175 5.67 5.67 0 013.036-1.158l.28-.014L18 18h1a2 2 0 001.995-1.85L21 16V8a2 2 0 00-1.85-1.995L19 6zm-1 7v2H6v-2h12zm0-4v2H6V9h12z"
                    fill="currentColor"
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Anna palautetta
            </span>
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <polygon
                    points="0 24 0 0 24 0 24 24"
                  />
                  <polygon
                    fill="currentColor"
                    points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                  />
                </g>
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
